### ISSUE TITLE
max_length for TextInput/PasswordInput [Feature Request]

### ISSUE THREAD

**Issue by Axle7XStriker** [beeware/toga#378]
## **Feature Request** 

I think we can have a max_length feature for the TextInput widget which fixes the maximum length of the text input. It is useful in case of making forms since in many forms there is boundations on the length of the text to be input by the user. Also such a feature is particularly useful in case of PasswordInput widget which is an extension of TextInput in cases where the user has to give a PIN or account number as an input which have a fixed length and in many other scenarios too.

**Comment by freakboy3742** [comment]
Sure - I can see how a "max_length" could be useful. A general facility for pre-validation (e.g., input can't contain numbers, or punctuation, or a dictionary word) could also be useful. While max_length is an obvious thing to pull out as a single argument, I'd be more interested in seeing this as part of a larger "validation and error reporting" feature for the widget.

Capturing a PIN is a separate issue, however - The broader Toga philosophy would dictate that a PIN input would be a different UI approach - as it requires a numeric keypad, and probably N separate input widgets (for an N digit pin) that are closely linked from a UI perspective.

**Comment by amitsin6h** [comment]
@freakboy3742 I would like to add this feature, it would be great if you could guide me, if required.

**Comment by taspotts** [comment]
### Some random thoughts on this:
#### When would the validation occur?
1.  As characters are entered into the TextInput?
2.  When the TextInput looses focus?
3.  Other?

#### How to handle a failed validation?
1.  Popup error message box?
2.  TextInput background goes red?
3.  Other?


#### Could the validators be "stackable" and/or can I role my own validator functions if a toga built-in doesn't exist?
Ex:
```python
c_input = toga.TextInput(validators=[toga.max_length(10), toga.numbers_only(), my_app.do_not_allow('7')])
```

**Comment by obulat** [comment]
It seems that you have fixed this issue, @saroad2 . Is there a reason for it to still be open?

**Comment by freakboy3742** [comment]
Closing on the basis it was addressed by #1088 and #1106.
